File: atcoder/abc171E_red_scarf.nim
# Vicfred
# https://atcoder.jp/contests/abc171/tasks/abc171_e
# bitwise
import strutils, sequtils, sugar

let n = stdin.readLine.parseInt
let a = stdin.readLine.split.map parseInt

var total = foldr(a, a xor b)

echo a.map((x) => x xor total).join(" ")

File: atcoder/abc171E_red_scarf.py
# Implementation by uninhm || Idea by Vicfred
# https://atcoder.jp/contests/abc171/tasks/abc171_e
# bitwise

input()
a = list(map(int, input().split()))

xor = a[0]
for i in a[1:]:
    xor ^= i

for i in a:
    print(xor ^ i, end=' ')
print()
